python_modules/dagster/dagster/_core/definitions/job_definition.py

```diff
@@ -292,6 +292,28 @@ def dagster_internal_init(
             _was_explicitly_provided_resources=_was_explicitly_provided_resources,
         )
 
+    @staticmethod
+    def for_external_job(
+        asset_keys: Iterable[AssetKey],
+        name: str,
+        metadata: Optional[Mapping[str, Any]] = None,
+        tags: Optional[Mapping[str, Any]] = None,
+    ) -> "JobDefinition":
+        from dagster._core.definitions import op
+
+        @op(name=f"{name}_op_inner")
+        def _op():
+            pass
```

Comment:
is this actually necessary? I think we can technically have empty graph definitions. If so, would recommend removing this for simplicity but it's fine if that's not possible.

Reply:
Unfortunately if the graph is empty we currently don't render the assets that exist on the job, which is why I had to add this. I'll add a comment explaining though